### PR TITLE
Mds 2337

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
This fixes the page and per page url params and makes the link sharable

### CODE REVIEW

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js [hunk 1]
```diff
@@ -110,20 +111,31 @@ export class IncidentsHomePage extends Component {
...
+        per_page: Strings.DEFAULT_PER_PAGE,
         region: formatParamStringToArray(this.params.region),
         major: this.params.major,
         search: this.params.search,
         year: this.params.year,
         incident_status: this.params.incident_status,
         codes: this.params.codes,
         determination: this.params.determination,
+        ...this.params,
       },
```
Comment (mmumby) [inline frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: what is left in this spread? everything should be defined in the state

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js [hunk 2]
```diff
@@ -110,20 +111,31 @@ export class IncidentsHomePage extends Component {
...
     };
   }
 
   componentDidMount() {
     const params = this.props.location.search;
-    this.renderDataFromURL(params);
+    if (params) {
+      this.renderDataFromURL(params);
+    } else {
+      const defaultParams = {
```
Comment (mmumby) [inline frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: Option 1) 
if there are no 'params' in the URL - then 'page' and 'per_page' are already defined in the state with their default when the page loads, so this latter step is not necessary

Option 2) 
the URL contains the params - if 'ComponentWillReceiveProps' is handling the params correctly, it should be setting the new values for 'page' and 'per_page'

File: frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js [hunk 3]
```diff
@@ -120,15 +123,22 @@ export class VarianceHomePage extends Component {
         issue_date_before: this.params.issue_date_before,
         expiry_date_before: this.params.expiry_date_before,
         expiry_date_after: this.params.expiry_date_after,
+        ...this.params,
       },
```
Comment (mmumby) [inline frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js]: same here, what is being spread? should be defined

File: frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js [hunk 4]
```diff
@@ -207,25 +217,26 @@ export class VarianceHomePage extends Component {
...
+      this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(updatedParams));
+      return { params: updatedParams };
+    });
   };
 
   handleVariancePageChange = (page, per_page) => {
     this.setState({ variancesLoaded: false });
     const params = { ...this.state.params, page, per_page };
+    this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(params));
     return this.props.fetchVariances(params).then(() => {
```
Comment (mmumby) [inline frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js]: `this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(params))` will cause `CompopnentWillReceiveProps' to send the new params to 'renderDataFromUrl` which will `fetchVariances`, this path will set the params correctly within the state as well as passing the correct params to the API

so, `fetchVariances` here is not needed.

File: frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js [hunk 5]
```diff
@@ -120,15 +123,22 @@ export class VarianceHomePage extends Component {
...
-
+    if (params) {
+      this.renderDataFromURL(params);
+    } else {
+      const defaultParams = {
+        page: Strings.DEFAULT_PAGE,
+        per_page: Strings.DEFAULT_PER_PAGE,
+      };
+      this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(defaultParams));
+    }
```
Comment (mmumby) [inline frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js]: same here ^

File: frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js (hunk 3 above)
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js]: removed

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js (hunk 1 above)
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: removed

File: frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js (hunk 3 above)
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js]: un removed

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js (hunk 2 above)
Comment (mmumby) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: @thegentlemanphysicist

**Pull request merged**